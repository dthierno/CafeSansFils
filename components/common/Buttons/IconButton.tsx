import { TouchableOpacity, StyleSheet, StyleProp, ViewStyle } from "react-native";
import { type LucideIcon, ShoppingBasket } from "lucide-react-native";

import COLORS from "@/constants/Colors";
import SPACING from "@/constants/Spacing";

type BasketButtonProps = {
  Icon?: LucideIcon;
  onPress?: () => void;
  accessibilityLabel?: string;
  style?: StyleProp<ViewStyle>,
};

export default function IconButton({
  style,
  Icon = ShoppingBasket,
  onPress,
  accessibilityLabel="Shopping Basket",
}: BasketButtonProps) {
  return (
    <TouchableOpacity
      style={[styles.iconButtonContainer, style]} 
      onPress={onPress}
      activeOpacity={0.5}
      testID="icon-button"
    >
      <Icon
        width={SPACING["xl"]}
        height={SPACING["xl"]}
        strokeWidth={2.2}
        color={COLORS.black}
        accessibilityLabel={accessibilityLabel}
      />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  iconButtonContainer: {
    backgroundColor: COLORS.lightGray,
    padding: SPACING.xs,
    borderRadius: 100,
  },
});
