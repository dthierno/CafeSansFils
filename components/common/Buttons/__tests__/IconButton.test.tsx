import { fireEvent, render, screen } from "@testing-library/react-native";

import IconButton from "../IconButton";
import { ShoppingBasket } from "lucide-react-native";

describe("IconButton", () => {
  const mockProps = {
    Icon: ShoppingBasket,
    onPress: jest.fn(),
    accessibilityLabel: "Shopping Basket",
  };

  it("renders the icon button", () => {
    render(<IconButton {...mockProps} />);
    const iconButton = screen.getByTestId("icon-button");
    expect(iconButton).toBeTruthy();
  });

  it("calls the onPress function when the icon button is pressed", () => {
    render(<IconButton {...mockProps} />);
    const iconButton = screen.getByTestId("icon-button");
    expect(iconButton).toBeTruthy();
    fireEvent.press(iconButton);
    expect(mockProps.onPress).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const tree = render(<IconButton {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
});