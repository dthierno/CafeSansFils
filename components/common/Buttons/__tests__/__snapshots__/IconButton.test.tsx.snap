// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#F2F2F7",
      "borderRadius": 100,
      "opacity": 1,
      "padding": 8,
    }
  }
  testID="icon-button"
>
  <RNSVGSvgView
    accessibilityLabel="Shopping Basket"
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="#000000"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2.2}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2.2}
    >
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="m15 11-1 9"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="m19 11-4-7"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="M2 11h20"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="m3.5 11 1.6 7.4a2 2 0 0 0 2 1.6h9.8a2 2 0 0 0 2-1.6l1.7-7.4"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="M4.5 15.5h15"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="m5 11 4-7"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
      <RNSVGPath
        accessibilityLabel="Shopping Basket"
        d="m9 11 1 9"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2.2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
